### DISCUSSION TITLE
How to encrypt an existing unencrypted database using sqlitemc shell?

### DISCUSSION THREAD

**Discussion by sheltongeosx** [utelle/SQLite3MultipleCiphers#58] [Q&A]
Hi, this is not an issue report but rather a newbie question - what are the exact command sequences to encrypt an existing non-encrypted database using sqlitemc shell ? I tried quite a few combinations of `pragma` statements but no success, It is greatly appreciated if you can provide with an example. Thank you very much in advance!

**Comment by utelle** [comment]
Since this is not an issue with the implementation, I'm going to convert this topic into a discussion.

The required SQL statements to encrypt an existing non-encrypted database depend on what exactly you want to accomplish.

For the default encryption scheme, usually _ChaCha20-Poly1305_, it is as simple as issuing a single pragma:

```SQL
PRAGMA rekey='passphrase';
```

If you want to use a different encryption scheme, you have to select it, before issuing the **rekey** pragma:

```SQL
PRAGMA cipher=aes256cbc;
PRAGMA rekey='passphrase';
```
selects the encryption scheme _wxSQLite3 AES 256-bit_.

If you want to use the _SQLCipher_ scheme in legacy mode (to be compatible with existing SQLCipher databases), you specify:

```SQL
PRAGMA cipher=sqlcipher;
PRAGMA legacy=4;
PRAGMA rekey='passphrase';
```

**Reply by sheltongeosx** [reply]
Thank you very much for your quick response! Yes it works! Further more, it behaves in the way that is more favorable to me - doing encryption in-place, instead of having users go through the route to create a new separate file for encrypting a non-encrypted file like SQLcipher does (using sqlcipher_export()).
